dbuf-lsp/src/navigation_handler/mod.rs

```diff
@@ -0,0 +1,139 @@
+//! Module aims to help with searches in dbuf files.
+//! Responses are easy to compute.
+//!
+//! Module should help with such requests:
+//! * (✓) `textDocument/definition`
+//! * (✓) `textDocument/typeDefinition`
+//! * (✓) `textDocument/hover
+//! * (✗!!) `textDocument/inlayHint` // for constructors type
+//!  
+//! Also it might be good idea to handle such requests:
+//!
+//! Perhaps, next time:
+//! * `inlayHint/resolve`
+//! * `textDocument/selectionRange`
+//! * `textDocument/moniker`
+//! * `textDocument/linkedEditingRange`
+//!
+//! These methods are also about navigation, but there no need to implement them:
+//! * `textDocument/prepareTypeHierarchy`
+//! * `typeHierarchy/supertypes`
+//! * `typeHierarchy/subtypes`
+//! * `textDocument/declaration`
+//! * `textDocument/implementation`
+//! * `textDocument/prepareCallHierarchy`
+//! * `callHierarchy/incomingCalls`
+//! * `callHierarchy/outgoingCalls`
+//! * `textDocument/documentLink`
+//! * `documentLink/resolve`
+//!
+
+mod hover;
+mod inlay_hint;
+mod navigation;
+
+use hover::get_hover;
+use navigation::find_definition;
+use navigation::find_type;
+
+use tower_lsp::jsonrpc::Result;
+use tower_lsp::lsp_types::request::*;
+use tower_lsp::lsp_types::OneOf::*;
+use tower_lsp::lsp_types::*;
+
+use crate::core::ast_access::WorkspaceAccess;
+use crate::core::navigator::Navigator;
+use crate::handler::Handler;
+
+pub struct NavigationHandler {}
+
+impl NavigationHandler {
+    /// `textDocument/definition` implementation.
+    ///
+    pub fn goto_definition(
+        &self,
+        access: &WorkspaceAccess,
+        pos: Position,
+        document: &Url,
+    ) -> Result<Option<GotoDefinitionResponse>> {
+        let file = access.read(document);
+        let navigator = Navigator::new(&file);
+
+        let symbol = navigator.get_symbol(pos);
+        let range = find_definition(&navigator, &symbol);
+
+        match range {
+            Some(range) => Ok(Some(GotoDefinitionResponse::Scalar(Location {
+                uri: document.to_owned(),
+                range,
+            }))),
+            None => Ok(None),
+        }
+    }
+
+    /// `textDocument/typeDefintion` implementation.
+    ///
+    pub fn goto_type_definition(
+        &self,
+        access: &WorkspaceAccess,
+        pos: Position,
+        document: &Url,
+    ) -> Result<Option<GotoTypeDefinitionResponse>> {
+        let file = access.read(document);
+        let navigator = Navigator::new(&file);
+
+        let symbol = navigator.get_symbol(pos);
+        let t = find_type(&navigator, symbol);
+
+        let range = find_definition(&navigator, &t);
+        range.map_or(Ok(None), |range| {
```

Comment:
So it's `Ok(range.map(|range| GotoTypeDefinitionResponse::Scalar(Location { ... })))`